sys/net/routing/rpl/of0.c

```diff
@@ -78,7 +78,7 @@ uint16_t calc_rank(rpl_parent_t *parent, uint16_t base_rank)
 /* We simply return the Parent with lower rank */
 rpl_parent_t *which_parent(rpl_parent_t *p1, rpl_parent_t *p2)
 {
-    if (p1->rank < p2->rank) {
+    if (p1->rank <= p2->rank) {
```

Comment:
I encountered the following use case: When using the `which_parent()` function I passed the current preferred parent into `p1` and the new parent into `p2`. This was more intuitive. In this case, when the ranks are equal, it is more efficient to return the old preferred parent instead of the new parent. Otherwise, the parents would alternate upon receiving DIO's from parents with equal ranks.